File: LeetCode/find-the-winning-player-in-coin-game/3222.py
class Solution:
    def winningPlayer(self, n: int, y: int) -> str:
        k=0
        while n>0 and y>=4:
            n=n-1
            y=y-4
            k+=1
        if(k%2==1):
            return "Alice"
        else:
            return "Bob"

File: LeetCode/find-the-winning-player-in-coin-game/3222.cpp
class Solution {
public:
    string winningPlayer(int x, int y) {
        int moves = 0;
        
        while (x > 0 && y >= 4) {
            x--;
            y -= 4;
            moves++;
        }

     return (moves % 2 == 1) ? "Alice" : "Bob";
    }
};